### PR TITLE
#4_July_27_EFCore_CodeFirst_approach

### PR DESCRIPTION
**Pull request by REtangler**

### CODE REVIEW

File: DeliveryService_EF/HwTasks/EfCoreCodeFirstTasks.cs [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        private readonly DataContext _db;
+
+        public EfCoreCodeFirstTasks(DataContext db)
+        {
+            _db = db;
+        }
+
+        public void AddProducts()
+        {
+            var products = new List<Product>
```
Comment (yevhenbilushko) [inline DeliveryService_EF/HwTasks/EfCoreCodeFirstTasks.cs:26]: move move products creation out of "AddProducts" method

File: DeliveryService_EF/HwTasks/EfCoreCodeFirstTasks.cs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            };
+
+            _db.Products.AddRange(products);
+            _db.SaveChanges();
+        }
+
+        public void GetProducts()
+        {
+            var products = _db.Products.ToList();
+            Console.WriteLine("Products in storage: " + products.Count);
```
Comment (yevhenbilushko) [inline DeliveryService_EF/HwTasks/EfCoreCodeFirstTasks.cs:133]: move all UI (Console.WriteLine) out of logic (repo and services)

File: DeliveryService_EF/HwTasks/EfCoreCodeFirstTasks.cs [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            {
+                product.Category = categories.SingleOrDefault(c => c.Id == product.CategoryId);
+                product.Supplier = suppliers.SingleOrDefault(s => s.Id == product.SupplierId);
+                Console.WriteLine($"{product.Name}, {product.Description}, {product.Type}, {product.Price}, {product.AmountInStock}, {product.Category?.Name}, {product.Supplier?.Name}");
+            }
+        }
+
+        public void AddCategories()
+        {
+            var categories = new List<Category>
```
Comment (yevhenbilushko) [inline DeliveryService_EF/HwTasks/EfCoreCodeFirstTasks.cs:147]: move
 var categories = new List<Category>
out of this method

**Review by yevhenbilushko — DISMISSED** [review]

**Comment by REtangler** [comment]
looks messy but all i did was fixed merge conflicts

**Review by yevhenbilushko — APPROVED** [review]

**Pull request merged**